doc/api/n-api.md

```diff
@@ -2993,6 +2994,74 @@ support it:
 * If the function is not available, provide an alternate implementation
   that does not use the function.
 
+## Promises
+
+Promises facilitate the construction of chains of asynchronous actions each of
+which hinges on the outcome of the previous.
+
+### napi_create_promise
+<!-- YAML
+added: v8.1.2
+-->
+```C
+NAPI_EXTERN
+napi_status napi_create_promise(napi_env env,
+                                napi_value* promise,
+                                napi_value* resolve_callback,
+                                napi_value* reject_callback);
+```
+
+- `[in] env`: The environment that the API is invoked under.
+- `[out] promise`: The resulting promise.
+- `[out] resolve_callback`: The JavaScript function to call in order to resolve
+the promise.
+- `[out] reject_callback`: The JavaScript function to call in order to reject
+the promise.
+
+Returns `napi_ok` if the API succeeded.
+
+This API creates a new Promise object and returns it in the first argument. In
+the second argument it returns a JavaScript function which, when called using
+`napi_call_function()`, will resolve the promise. Similarly, in the third
+argument it returns a JavaScript function which, when called using
+`napi_call_function()`, will reject the promise. For example:
+
+```C
+napi_status status;
+napi_value promise, resolve, reject, resolution;
+
+status = napi_create_promise(env, &promise, &resolve, &reject);
+if (status != napi_ok) return;
+
+status = napi_create_number(env, 42, &resolution);
+if (status != napi_ok) return;
+
+status = napi_call_function(env, promise, resolve, 1, &resolution, NULL);
+if (status != napi_ok) return;
+
+// The promise is now resolved to the value of resolution
+```
+
+### napi_is_promise
+<!-- YAML
+added: v8.1.2
+-->
+```C
+NAPI_EXTERN napi_status napi_is_promise(napi_env env,
+                                        napi_value value,
+                                        bool* result);
+```
+
+- `[in] env`: The environment that the API is invoked under.
+- `[in] value`: The value for which to check whether it is a promise.
+- `[out] result`: Whether the passed-in value is a promise.
+
+Returns `napi_ok` if the API succeeded.
+
+This API checks whether a passed-in value is a promise and returns the result
```

Comment:
I think it might be worth mentioning that this checks whether it is a *built-in* native Promise, not just any promise object that implements the Promises/A+ interface